### PR TITLE
fix: More early returns

### PR DESCRIPTION
**Pull request by jalil-salame**
I found two places that could use an early return to simplify the code.

### CODE REVIEW

File: src/vec.rs [hunk 1]
```diff
@@ -130,16 +130,16 @@ impl<T> EcoVec<T> {
     /// Create a new, empty vec with at least the specified capacity.
     #[inline]
     pub fn with_capacity(capacity: usize) -> Self {
-        let mut vec = Self::new();
```
Comment (laurmaedje) [inline src/vec.rs:133]: I actually think the previous version was more readable here.

File: src/vec.rs [hunk 2]
```diff
@@ -1140,27 +1140,29 @@ impl<T: Clone> ExactSizeIterator for IntoIter<T> {}
 
 impl<T> Drop for IntoIter<T> {
     fn drop(&mut self) {
-        if self.unique && self.vec.is_allocated() {
+        if !(self.unique && self.vec.is_allocated()) {
```
Comment (laurmaedje) [inline src/vec.rs:1143]: Here I agree that the early return is better. But could you De-Morgan the condition to `!self.unique || !self.vec.is_allocated()`?

Comment (jalil-salame) [reply src/vec.rs]: Resolved in latest commit

**Comment by laurmaedje** [comment]
Thanks again!

**Pull request merged**